src/javascript/app/common/traffic_source.js

```diff
@@ -48,12 +49,23 @@ const TrafficSource = (() => {
         const params         = Url.paramsHash();
         const param_keys     = ['utm_source', 'utm_medium', 'utm_campaign'];
 
-        if (params.utm_source) { // url params can be stored only if utm_source is available
-            param_keys.map((key) => {
-                if (params[key] && !current_values[key]) {
-                    cookie.set(key, params[key], { sameSite: 'none', secure: true });
-                }
-            });
+        // When the user comes to the site with URL params
+        if (params.utm_source || params.utm_medium || params.utm_campaign) {
+
+            // if url is missing one of required fields, do nothing
+            const is_need_change = param_keys.every((param) => param in params);
+            
+            if (is_need_change) {
+                param_keys.forEach((key) => {
+                    if (params[key]) {
+                        cookie.set(key, params[key], { sameSite: 'none', secure: true });
+                    }
+                });
+            }
+            
+        } else if (!Cookies.get('utm_data')) {
+            initCookie();
```

Comment:
any reason why do we need initCookie() again here?, since have called initCookie on line 47